src/bindgen/parser.rs

```diff
@@ -533,13 +533,15 @@ impl Parse {
                     self.load_syn_ty(crate_name, mod_cfg, item);
                 }
                 syn::Item::Impl(ref item_impl) => {
-                    let has_assoc_const = item_impl
-                        .items
-                        .iter()
-                        .any(|item| matches!(item, syn::ImplItem::Const(_)));
-                    if has_assoc_const {
-                        impls_with_assoc_consts.push(item_impl);
-                    }
+                    let mut impls_with_assoc_ty = Vec::new();
+
+                    item_impl.items.iter().for_each(|item| {
+                        if matches!(item, syn::ImplItem::Const(_)) {
+                            impls_with_assoc_consts.push(item_impl);
+                        } else if let syn::ImplItem::Type(t) = item {
+                            impls_with_assoc_ty.push((&t.ident, &t.ty));
```

Comment:
This array seems wrong, it's not storing `impls`, but ident and types, so maybe just call the vector `associated_tys` or so?